### DISCUSSION TITLE
Callbacks when wallet is connected

### DISCUSSION THREAD

**Discussion by kemilandmeta** [use-cardano/use-cardano#97] [Q&A] [answered]
How we could inject a callback when the wallet gets connected.

**Comment by GGAlanSmithee** [comment]
hey @kemilandmeta thanks for your comment.

I think this is technically possible with the current API, since you can "listen" to changes to a number of values that would indicate connection to wallet.

For example:

```ts
const { walletApi } = useCardano()

useEffect(() => {
  if (!walletApi) return

  // wallet connected, do something
}, [walletApi])
```

However, this isn't guaranteed to only execute once, which is what I guess you are after, so I think adding a callback like this to the API makes sense.

What is the expected behaviour of this callback in your use case?

* Should this only happen once, as the first wallet connection gets connected, or any additional connection? (ie, wallet provider changes, account changes, etc)
* should this happen after the underlying `lucid` instance is initialized, so it's available for use?

**Reply by kemilandmeta** [reply]
This is great, will try it out and let you know,

We need to redirect to the app main page once a user connects the wallet. 

We should also consider wallet changes. 

Not required for lucid init we validate lucid is initiated when it's required.

**Reply by GGAlanSmithee** [reply]
ok, I will implement this over the weekend!

**Reply by kemilandmeta** [reply, marked as answer]
Use effect did the trick 

```ts
useEffect(() => {
    if(!account.address) {
      console.log("wallet disconnected");
      // do something
      return;
    }
    console.log("wallet connected");
    // do something 
    
    }, [account]);
```

**Reply by kemilandmeta** [reply]
Any advice on how to maintain the wallet state between pages?

**Comment by GGAlanSmithee** [comment]
Hi @kemilandmeta sorry, I missed that you had more questions.

Since use-cardano uses a React.Context under the hood, state should (could) be shared across your entire application. Just make sure that you wrap all pages you want to share state across with the same `<CardanoProvider>`

**Reply by kemilandmeta** [reply]
It works perfectly, a great, and is a stable component.

**Comment by kemilandmeta** [comment]
Thanks a lot again for the support

**Reply by GGAlanSmithee** [reply]
My pleasure, thansk for using the package 🙇🏼
